extensions-contrib/opentelemetry-extensions/src/test/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReaderTest.java

```diff
@@ -0,0 +1,343 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.opentelemetry.protobuf;
+
+import com.google.common.collect.ImmutableList;
+import io.opentelemetry.proto.common.v1.AnyValue;
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.metrics.v1.Metric;
+import io.opentelemetry.proto.metrics.v1.MetricsData;
+import org.apache.druid.data.input.InputRow;
+import org.apache.druid.data.input.impl.ByteEntity;
+import org.apache.druid.data.input.impl.DimensionsSpec;
+import org.apache.druid.data.input.impl.StringDimensionSchema;
+import org.apache.druid.java.util.common.parsers.CloseableIterator;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OpenTelemetryMetricsProtobufReaderTest
+{
+  private static final long TIMESTAMP = System.nanoTime();
+  public static final String RESOURCE_ATTRIBUTE_COUNTRY = "country";
+  public static final String RESOURCE_ATTRIBUTE_VALUE_USA = "usa";
+
+  public static final String RESOURCE_ATTRIBUTE_ENV = "env";
+  public static final String RESOURCE_ATTRIBUTE_VALUE_DEVEL = "devel";
+
+  public static final String INSTRUMENTATION_LIBRARY_NAME = "mock-instr-lib";
+  public static final String INSTRUMENTATION_LIBRARY_VERSION = "1.0";
+
+  public static final String METRIC_ATTRIBUTE_COLOR = "color";
+  public static final String METRIC_ATTRIBUTE_VALUE_RED = "red";
+
+  public static final String METRIC_ATTRIBUTE_FOO_KEY = "foo_key";
+  public static final String METRIC_ATTRIBUTE_FOO_VAL = "foo_value";
+
+  private final MetricsData.Builder requestBuilder = MetricsData.newBuilder();
+
+  private final Metric.Builder metricBuilder = requestBuilder.addResourceMetricsBuilder()
+          .addInstrumentationLibraryMetricsBuilder()
+          .addMetricsBuilder();
+
+  private final DimensionsSpec dimensionsSpec = new DimensionsSpec(ImmutableList.of(
+          new StringDimensionSchema("descriptor." + METRIC_ATTRIBUTE_COLOR),
+          new StringDimensionSchema("descriptor." + METRIC_ATTRIBUTE_FOO_KEY),
+          new StringDimensionSchema("custom." + RESOURCE_ATTRIBUTE_ENV),
+          new StringDimensionSchema("custom." + RESOURCE_ATTRIBUTE_COUNTRY)
+  ), null, null);
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Before
+  public void setUp()
+  {
+    requestBuilder
+            .getResourceMetricsBuilder(0)
```

Comment:
I think this reads better (I don't think there's a difference in semantics)
```suggestion
            .addResourceMetricsBuilder()
```

Reply:
IIUC they're different. because the `requestBuilder` (now called the metricsDataBuilder) is created at the same time as the test class and used to derive the `metricsBuilder, ` .getResourceMetricsBuilder(0) is used to get the same `ResourceMetrics.Builder` that the `metricsBuilder` exists within. Calling `.addResourceMetricsBuilder()` would create a new ResourceMetrics.Builder and add it to a list of ResourceMetrics.Builders that exist within the `metricsDataBuilder`, alongside the one that houses the `metricsBuilder` but would not reference it to the same one. This is the same model that we follow with the [OpenCensusProtobufConverter](https://github.com/confluentinc/opencensus-protobuf-converter/blob/49c85d0f7d444fca2bfef3e1770afcfb90392786/src/test/java/io/confluent/connect/opentelemetry/OpenTelemetryMetricsConverterTest.java#L85)

Reply:
Makes sense.  Thanks for the detailed explanation.